addons/web/static/src/core/assets.js

```diff
@@ -0,0 +1,185 @@
+        }
+
+        loadedBundleTemplates[name] = {
+            legacyTemplates: Array.from(doc.querySelectorAll("templates")),
+            owlTemplates: owlTemplates,
+        };
+    }
+    return loadedBundleTemplates[name];
+}
+
+//------------------------------------------------------------------------------
+// Exports
+//------------------------------------------------------------------------------
+
+/**
+ * Loads the given assets. Currently, when passing bundles, only the templates
+ * key is supported, as loading a bundle's JS/CSS asynchronously requires some
+ * groundwork.
+ *
+ * @param {Object} assets
+ * @param {Bundle[]} [assets.bundles] description of the bundles to load
+ * @param {string[]} [assets.jsLibs] urls of the javascript libraries to load
+ * @param {string[]} [assets.cssLibs] urls of the css libraries to load
+ * @returns {Promise<{bundles: LoadedBundle[]}>} An object describing the loaded
+ *      assets. The js and css libs are loaded globally and will be loaded when
+ *      this promise is resolved.
+ */
+export async function loadAssets(assets) {
+    const proms = [];
+    const loadedAssets = {};
+    if ('bundles' in assets) {
+        loadedAssets.bundles = {};
+        assets.bundles.forEach(({ name, js, css, templates }) => {
+            loadedAssets.bundles[name] = {};
+            if (templates) {
+                proms.push(loadBundleTemplates(name).then(loadedTemplates => {
+                    loadedAssets.bundles[name].templates = loadedTemplates;
+                }));
+            }
+            // TODO: if ('js/css' in assets) {...}
+        });
+    }
+    if ('jsLibs' in assets) {
+        proms.push(...assets.jsLibs.map(loadJS));
+    }
+    if ('cssLibs' in assets) {
+        proms.push(...assets.cssLibs.map(loadCSS));
+    }
+    await Promise.all(proms);
+    return loadedAssets;
+}
+/**
+ * Loads the given assets, and adds the loaded owl templates into the current
+ * environment's qweb instance.
+ *
+ * @param {Object} assets
+ * @param {Bundle[]} [assets.bundles] the bundles to load
+ */
+export function useAssets(assets) {
+    const env = useEnv();
+    onWillStart(async () => {
+        const owlTemplates = [];
+        const loadedAssets = await loadAssets(assets);
+        const { bundles } = loadedAssets;
+        if (bundles) {
+            Object.values(bundles).forEach(({ templates }) => {
+                owlTemplates.push(...templates.owlTemplates);
+            });
+        }
+        // Some templates might already be defined by another bundle, but need
+        // to be included in the current bundle for primary inherits to work.
+        // We don't want to add those again.
+        const toAdd = owlTemplates.filter(template => {
+            const templateName = template.getAttribute('t-name');
+            return !(templateName in env.qweb.templates);
+        });
+        env.qweb.addTemplates(`<templates>${toAdd.map(t => t.outerHTML).join('\n')}</templates>`);
```

Comment:
it may be kind of duplicated with the load template code from web client. let's talk about this tomorrow